plugins/DNS/SiteManagerPlugin.py

```diff
@@ -0,0 +1,34 @@
+from Config import config
+from Plugin import PluginManager
+
+from .DNSResolver import DNSResolver
+
+allow_reload = False
+
+@PluginManager.registerTo('SiteManager')
+class SiteManagerPlugin:
+    _dns_resolver = None
+
+    @property
+    def dns_resolver(self):
+        if not self._dns_resolver:
+            nameservers = config.dns_nameservers
+            configure = config.dns_configure
+
+            self._dns_resolver = DNSResolver(
+                site_manager=self,
+                nameservers=nameservers,
+                configure=configure,
```

Comment:
Dangling comma.

Reply:
Is this really needed to be fixed? Python allows trailing commas in lists and function calls.

Does ZeroNet follow any coding standard? Because according to [PEP 8](https://www.python.org/dev/peps/pep-0008/#when-to-use-trailing-commas), trailing comma here should be correct.

Reply:
ZeroNet doesn't really follow PEP 8. I think that following the existing code is better than following a standard. You can take a look at many other places in ZeroNet code where the dangling comma is not added. So even if it's not the correct way, it's at least the *same* way.

Reply:
> I think that following the existing code is better than following a standard.

Yes, But I didn't check all ZeroNet code standards so I just used some things from PEP 8.

> So even if it's not the correct way, it's at least the same way.

I agree. I will remove this comma, but in the future, it would be good to reformat ZeroNet code to some known Python coding standard.

Reply:
Actually, flake8 *is* using PEP 8 but some rules are ignored, so we're nearby.

Reply:
> Should dangling comma only not be added in function calls, or also not in multi-line lists?

I'm not sure about this one, let me check in a sec.

Reply:
@imachug What should be used for multi-line lists? Should there be dangling comma or not?

Reply:
We don't use dangling commas at all. Just a few examples to support this:

```python
casubjects = [
    "/C=US/O=Amazon/OU=Server CA 1B/CN=Amazon",
    "/C=US/O=Let's Encrypt/CN=Let's Encrypt Authority X3",
    "/C=US/O=DigiCert Inc/OU=www.digicert.com/CN=DigiCert SHA2 High Assurance Server CA",
    "/C=GB/ST=Greater Manchester/L=Salford/O=COMODO CA Limited/CN=COMODO RSA Domain Validation Secure Server CA"
]
```

(and a few more from CryptConnection). Also:

```python
return self.render(
    "src/Ui/template/wrapper.html",
    server_url=server_url,
    inner_path=inner_path,
    file_url=re.escape(file_url),
    file_inner_path=re.escape(file_inner_path),
    address=site.address,
    title=html.escape(title),
    body_style=body_style,
    meta_tags=meta_tags,
    query_string=re.escape(inner_query_string),
    wrapper_key=site.settings["wrapper_key"],
    ajax_key=site.settings["ajax_key"],
    wrapper_nonce=wrapper_nonce,
    postmessage_nonce_security=postmessage_nonce_security,
    permissions=json.dumps(site.settings["permissions"]),
    show_loadingscreen=json.dumps(show_loadingscreen),
    sandbox_permissions=sandbox_permissions,
    rev=config.rev,
    lang=config.language,
    homepage=homepage,
    themeclass=themeclass,
    script_nonce=script_nonce
)
```

Reply:
Also fixed!